### PR TITLE
Go to relevant Summary if available after edit

### PR DESCRIPTION
**Pull request by dwyeradam**
### What is the context of this PR?
Upon returning to a question in any fashion (via edit link, using back buttons or entering question url) when submitting that question if a Summary page (Final or Section) is still available (no change in routing or dependencies) for it then it should return to the summary page.

### How to review 
Use existing test_section_summary.json for testing of returns to both section and final summaries. Final summary can be made unavailable with a change of routing.
Use test_dependencies schemas to show behaviour when summary is unavailable due to a dependency change.

### CODE REVIEW

File: app/questionnaire/path_finder.py
Comment (shahi645) [inline app/questionnaire/path_finder.py]: We are assuming here that all schemas will either finish with summary or confirmation. Just want to check that will always be the case?

File: app/questionnaire/questionnaire_schema.py
Comment (nealedj) [inline app/questionnaire/questionnaire_schema.py:52]: The `piped-answers-in-repeating-groups-fix` branch patches a `parent_id` onto blocks and groups. Once that's merged that may by a better solution for finding the parent section.

I.e. (in `PathFinder.get_next_location`)
```
location_group = self.schema.get_group(current_block_id['parent_id'])
location_section = self.schema.get_section(location_group['parent_id'])
```

If we merge this first then consider altering this in the other branch. @dwyeradam

File: app/questionnaire/path_finder.py
Comment (MIslam92) [inline app/questionnaire/path_finder.py]: Looking at how you returned the section_summary_location below and to make it more readable/consistent  would it be worth declaring a variable called summary_location and returning that instead of `routing_path[-1]` plus you could remove the comments as it would be self explanatory from the name alone

File: app/questionnaire/path_finder.py
Comment (nealedj) [reply app/questionnaire/path_finder.py]: For every schema currently in the repo this is true. I think we depend on this being the case in other places as well as this.

Let's check that validator enforces this.

Comment (nealedj) [reply app/questionnaire/path_finder.py]: I just checked - [it does](https://github.com/ONSdigital/eq-schema-validator/blob/master/app/validation/validator.py#L212).

File: app/questionnaire/path_finder.py
Comment (nealedj) [inline app/questionnaire/path_finder.py]: Rather than introducing a dependency on a global (that then needs to be mocked) could we pass `completed_blocks` in the `PathFinder` constructor?

File: app/questionnaire/path_finder.py
Comment (shahi645) [reply app/questionnaire/path_finder.py]: Makes sense just wanted to check that we should enforce this

**Review by MIslam92 — CHANGES_REQUESTED** [review]
Minor nit pick?

**Review by nealedj — CHANGES_REQUESTED** [review]

File: app/questionnaire/path_finder.py
Comment (dwyeradam) [reply app/questionnaire/path_finder.py]: @nealedj, Yes, or it may be better to actually replace answer_store with questionnaire_store as we can get both completed_blocks and answer_store from that.

File: app/questionnaire/path_finder.py
Comment (dwyeradam) [reply app/questionnaire/path_finder.py]: Yes the last block should be a summary or confirmation page. Regardless the last block will never be completed at this point otherwise the questionnaire would already have been submitted.

File: app/questionnaire/questionnaire_schema.py
Comment (dwyeradam) [reply app/questionnaire/questionnaire_schema.py:52]: I'll have a look at that then.

File: app/questionnaire/path_finder.py
Comment (dwyeradam) [reply app/questionnaire/path_finder.py]: I don't mind doing that but i'm not sure how more readable that is. 
@nealedj, @shahi645  thumbs up for change, thumbs down for not required please.
It would never be the same as the declaration of variable `section_summary_location` as that is set by a method. That method could then return None and if so I wouldn't return `section_summary_location` at all. Here we would always return the summary location so it's just creating a variable to immediately return it.

File: app/questionnaire/path_finder.py
Comment (nealedj) [reply app/questionnaire/path_finder.py]: Yeah that's fair - I guess the only disadvantage might be ease of testing. I figure that the `Navigation` object accepts those arguments separately for that reason.

File: app/questionnaire/path_finder.py
Comment (nealedj) [reply app/questionnaire/path_finder.py]: I'll say no in this case. Although assigning values to variables in order to help add context is a valid point.

It's a reasonable implicit bit of logic that if the survey is complete then the next location must be the last (final) location.

File: app/questionnaire/path_finder.py
Comment (dwyeradam) [reply app/questionnaire/path_finder.py]: Actually I think i'll follow `Navigation's` lead and pass it in. Simpler all round as easier for testing and checking use of those variables across path_finder.

**Review by shahi645 — APPROVED** [review]
Good feature we are adding 👍

**Review by nealedj — APPROVED** [review]

**Comment by dwyeradam** [comment]
@MIslam92, can you have another look please and approve if happy. Or raise a new comment if unhappy. Despite the fact I have two reviews don't just dismiss yours as you're committed to a review now.

**Review by MIslam92 — APPROVED** [review]
👍 "Squishy Squashy"

**Pull request merged**
